src/posit/connect/external/databricks.py

```diff
@@ -0,0 +1,60 @@
+import abc
+from typing import Callable, Dict, Optional
+
+from ..client import Client, is_local
+from ..oauth import OAuthIntegration
+
+HeaderFactory = Callable[[], Dict[str, str]]
+
+# https://github.com/databricks/databricks-sdk-py/blob/v0.20.0/databricks/sdk/credentials_provider.py
+# https://github.com/databricks/databricks-sql-python/blob/v3.1.0/src/databricks/sql/auth/authenticators.py
+# In order to keep compatibility with the Databricks SDK
+class CredentialsProvider(abc.ABC):
```

Comment:
Why not import https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/credentials_provider.py#L30 directly?

Reply:
I wanted to avoid adding their SDK as a dependency. I'll test and see if we can remove this abstract definition altogether.

Reply:
Let's leave this for now. I think we will have a similar problem for our content types. It would be nice to provide helpers for each framework when obtaining the identity token from the headers but it will vary by content type.